File: Pset2/exercises/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

float breakLetters(string text);

int main(void)
{

    string text = get_string("Digite o texto: ");
    float result;
    int grade;

    result = breakLetters(text);

    grade = round(result);

    if(grade < 1)
    {

        printf("Before Grade 1\n");

    }
    else if(grade > 16)
    {

        printf("Grade 16+\n");

    }
    else
    {

        printf("Grade %i\n",grade);

    }

}

float breakLetters(string text)
{

    int counterOfLetters = 0, countOfWords = 0, countOfSentences = 0;
    float index;

    for(int i = 0, n = strlen(text); i < n; i++)
    {

        if(text[i] != ' ' && text[i] != '.' && text[i] != ',' && text[i] != '?' && text[i] != '!' && text[i] != 39){

            counterOfLetters += 1;

            if(i == n-1){
                countOfSentences += 1;
                countOfWords += 1;
            }


        }
        else
        {

            if(text[i] == '.' || text[i] == '!' || text[i] == '?'){

                countOfSentences += 1;
                countOfWords += 1;

            }
            else if(text[i-1] != '.' && text[i-1] != '!' && text[i-1] != '?' && text[i-1] != ',' && text[i] != 39)
            {

                countOfWords += 1;

            }

        }

    }

    index = 0.0588 * ((counterOfLetters / (float) countOfWords)*100) - 0.296 * ((countOfSentences / (float) countOfWords)*100) - 15.8;

    return index;

}

File: Pset6/readability.py
def break_letters(text):
    counter_of_letters = 0
    count_of_words = 0
    count_of_sentences = 0

    for char in text:
        if char.isalpha():
            counter_of_letters += 1
            if char == '.' or char == '!' or char == '?':
                count_of_sentences += 1
            if char == ' ':
                count_of_words += 1
        elif char in ['.', '!', '?']:
            count_of_sentences += 1
            count_of_words += 1

    if count_of_words == 0:
        return 0.0

    index = 0.0588 * (counter_of_letters / count_of_words * 100) - 0.296 * (count_of_sentences / count_of_words * 100) - 15.8

    return index

def main():
    text = input("Digite o texto: ")
    result = break_letters(text)
    grade = round(result)

    if grade < 1:
        print("Before Grade 1")
    elif grade > 16:
        print("Grade 16+")
    else:
        print(f"Grade {grade}")

main()